src/edgeitem.cpp

```diff
@@ -86,3 +105,22 @@ bool EdgeItem::hasGrip(GripItem* grip) const
 {
   return grip == left || grip == right;
 }
+
+QPointF EdgeItem::nearestPointOnLine(const QPointF& point)
+{
+  QLineF norm = line().normalVector();
+  norm.setPoints(point, point + QPointF(norm.dx(), norm.dy()));
+  QPointF result;
+  norm.intersects(line(), &result);
```

Comment:
By the way, do you understand this syntax?

Reply:
Yes. If I remember correctly &result is so that whatever norm.intersetcs() does within the function is returned back to result.

Reply:
That's the _intent_ of it, but do you understand how it works?